### PR TITLE
layers: Only read relevant settings

### PR DESCRIPTION
**Pull request by christophe-lunarg**
Layers settings using environment variables on Android is slow. This is causing some C.I. performance issues.

To resolve this, we are now reading only settings which dependences are met.

Also, if legacy 'enables' and 'disables' settings are set, then we don't read the new settings.

WIP: Adding some unit tests on utility functions

### CODE REVIEW

File: layers/CMakeLists.txt
Comment (juan-lunarg) [inline layers/CMakeLists.txt:1]: Will need to update Android.mk as well

**Review by juan-lunarg — DISMISSED** [review]
Update Android.mk build as well.

File: layers/layer_options.cpp [hunk 1]
```diff
@@ -470,40 +455,55 @@ void ProcessConfigAndEnvSettings(ConfigAndEnvSettings *settings_data) {
...
+    SetLocalEnableSetting(config_enable_values, ",", settings_data->enables);
+    SetLocalEnableSetting(envvar_enable_values, env_delimiter, settings_data->enables);
+
+    const std::string config_disable_values = GetConfigValue(SETTING_DISABLES);
+    const std::string envvar_disable_values = GetEnvVarValue(SETTING_DISABLES);
+    SetLocalDisableSetting(config_disable_values, ",", settings_data->disables);
+    SetLocalDisableSetting(envvar_disable_values, env_delimiter, settings_data->disables);
+
+    // When legacy enable/disable settings are not used, check whether the new settings are used
+    if (config_enable_values.empty() && envvar_enable_values.empty() && config_disable_values.empty() &&
```
Comment (juan-lunarg) [inline layers/layer_options.cpp:470]: Worth addressing the Android performance issue being fixed

File: layers/layer_settings_utils.h [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+        }
+    }
+    return token;
+}
+
+static inline std::string FindDelimiter(const std::string &setting_values) {
+    if (setting_values.find(";")) {
+        return ";"; // Typically WIN32 env var
+    } else if (setting_values.find(":")) {
+        return ":"; // Typically UNIX env var
```
[the remark this answers is not in the document]
Comment (christophe-lunarg) [reply layers/layer_settings_utils.h:54]: Fixed

File: tests/positive/layer_settings_utils.cpp [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+#include "../framework/layer_validation_tests.h"
+#include "layer_settings_utils.h"
+
+class PositiveLayerSettingUtils : public VkPositiveLayerTest {};
+
+// These test check utils in the layer without needing to create a full Vulkan instance
```
[the remark this answers is not in the document]
Comment (christophe-lunarg) [reply tests/positive/layer_settings_utils.cpp:18]: Fixed

This said, others tests use the same approach,

**Pull request closed without merging**